lib/kafka_ex/protocol/produce.ex

```diff
@@ -32,41 +32,45 @@ defmodule KafkaEx.Protocol.Produce do
   end
 
   def create_request(correlation_id, client_id, %Request{topic: topic, partition: partition, required_acks: required_acks, timeout: timeout, compression: compression, messages: messages}) do
-    message_set = create_message_set(messages, compression)
-    KafkaEx.Protocol.create_request(:produce, correlation_id, client_id) <>
-      << required_acks :: 16-signed, timeout :: 32-signed, 1 :: 32-signed >> <>
-      << byte_size(topic) :: 16-signed, topic :: binary, 1 :: 32-signed, partition :: 32-signed, byte_size(message_set) :: 32-signed >> <> message_set
+    {message_set, mssize} = create_message_set(messages, compression)
+    [KafkaEx.Protocol.create_request(:produce, correlation_id, client_id),
+      << required_acks :: 16-signed, timeout :: 32-signed, 1 :: 32-signed >>,
+      << byte_size(topic) :: 16-signed, topic :: binary, 1 :: 32-signed, partition :: 32-signed, mssize :: 32-signed >>,
+      message_set]
   end
 
   def parse_response(<< _correlation_id :: 32-signed, num_topics :: 32-signed, rest :: binary >>), do: parse_topics(num_topics, rest, __MODULE__)
   def parse_response(unknown), do: unknown
 
-  defp create_message_set([], _compression_type), do: ""
+  defp create_message_set([], _compression_type), do: {"", 0}
   defp create_message_set([%Message{key: key, value: value}|messages], :none) do
-    message = create_message(value, key)
-    message_set = << 0 :: 64-signed >> <> << byte_size(message) :: 32-signed >> <> message
-    message_set <> create_message_set(messages, :none)
+    {message, msize} = create_message(value, key)
+    message_set = [<< 0 :: 64-signed >>, << msize :: 32-signed >>, message]
+    {message_set2, ms2size} = create_message_set(messages, :none)
+    {[message_set, message_set2], 8 + 4 + msize + ms2size}
```

Comment:
It would be preferable if the bare numbers (8 & 4) here and in the rest of this function, and in `create_message/3` and `bytes/1` were module attributes so that it's more clear why the message set size is being padded with them.